### PR TITLE
Add free resources from CodeCrafters

### PR DESCRIPTION
**Pull request by abhishekbose87**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo
Add resources to free-programming-interactive-tutorials

### Description
We help engineers get really good at their craft by building your own Redis, Git, Docker, SQLite from scratch.
We wanted to add some content that is free on our website.

### Why is this valuable (or not)?
valuable for experienced folks who want to dig deeper into their dev tools. 

### How do we know it's really free?
it can be accessed without creating an account

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: more/free-programming-interactive-tutorials-en.md [hunk 1]
```diff
@@ -161,6 +161,7 @@
 * [Go Tutorial](https://www.w3schools.com/go/) - W3Schools
 * [Learn Go](https://www.codecademy.com/learn/learn-go) - Google, Robert Griesemer, Rob Pike, and Ken Thompson (Codecademy) *(account required)*
 * [Start using Go](https://docs.microsoft.com/en-us/learn/paths/go-first-steps/) - Microsoft
+* [TCP Servers in Go](https://app.codecrafters.io/concepts/go-tcp-server) - CodeCrafters *(no account required)*
```
Comment (eshellman) [inline more/free-programming-interactive-tutorials-en.md:164]:
```suggestion
* [TCP Servers in Go](https://app.codecrafters.io/concepts/go-tcp-server) - CodeCrafters
```
Same issue with the other 2

Comment (abhishekbose87) [reply more/free-programming-interactive-tutorials-en.md]: @eshellman I have removed the required texts. Hope this PR looks good now.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by syedmaazsaeed** [comment]
@eshellman  If you're still facing any MERN stack issues, can I help ?

**Pull request merged**